torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -976,6 +978,33 @@ def test_average_parameters(self):
                 for p in model.parameters():
                     self.assertEqual(p.data, torch.ones_like(p.data) * 0.5)
 
+        @unittest.skipIf(
+            BACKEND != "nccl" and BACKEND != "gloo",
+            "MPI backend does not support creating subgroups on CUDA devices",
+        )
+        @skip_if_lt_x_gpu(2)
+        @skip_if_odd_num_of_gpus()
```

Comment:
Why does this not work if there is a world size of 5 for example? Can we enhance the test to make it work regardless of world size?

Reply:
Yeah, actually I forgot to label this PR [WIP]. I plan to remove `skip_if_odd_num_of_gpus` decorator and simplify the test code.